custom_components/miele/miele_at_home.py

```diff
@@ -77,7 +78,10 @@ async def action(self, device_id, body):
             result = await self.hass.async_add_executor_job(func)
             if result.status_code == 401:
                 _LOGGER.info("Request unauthorized - attempting token refresh")
-                if self._session.refresh_token():
+                self._session._delete_token()
```

Comment:
Wouldn't this mean that even if a token could be refreshed automatically everyone will always need manually refresh it? If so this should only be done when the refreshing fails.

Reply:
I made a change so that there's only a manual refresh if the session still isn't authorized after the refresh. What do you think?

Reply:
I will try to check it out later today, thanks for your effort! Sounds like a good solution :-)